tests/test_project_time.py

```diff
@@ -78,38 +78,43 @@ def test_convert_to_project_time(registry_mgr):
         raw_data=tempo_data_with_tz,
         converted_data=tempo_data_mapped_time,
     )
-    compare_time_conversion(
-        resstock_time_dim, project_time_dim, wrap_time=False, expect_error=True
-    )
-    compare_time_conversion(
-        comstock_time_dim, project_time_dim, wrap_time=False, expect_error=True
-    )
-    compare_time_conversion(
-        resstock_time_dim, project_time_dim, wrap_time=True, expect_error=False
-    )
-    compare_time_conversion(
-        comstock_time_dim, project_time_dim, wrap_time=True, expect_error=False
-    )
+
+    # [2.1] test time-wrap when dataset and project have different time interval type
+    # expect no error with original dataset and project time configs
+    compare_time_conversion(resstock_time_dim, project_time_dim, expect_error=False)
+    compare_time_conversion(comstock_time_dim, project_time_dim, expect_error=False)
     compare_time_conversion(
         tempo_time_dim, project_time_dim, df=tempo_data_mapped_time, expect_error=False
     )
+    project_time_dim.model.wrap_time_allowed = False
+    compare_time_conversion(resstock_time_dim, project_time_dim, expect_error=True)
+    compare_time_conversion(comstock_time_dim, project_time_dim, expect_error=True)
```

Comment:
nit: Add optional kwarg that provides a text snippet that should be in the expected error?

Reply:
The error msg is here: https://github.com/dsgrid/dsgrid/blob/ae76c0814d59652ce213f7a563bbff51238268f8/tests/test_project_time.py#L182-L184